src/bootstrap/tool.rs

```diff
@@ -607,7 +607,14 @@ macro_rules! tool_extended {
 
             fn should_run(run: ShouldRun<'_>) -> ShouldRun<'_> {
                 let builder = run.builder;
-                run.path($path).default_condition(builder.config.extended)
+                run.path($path).default_condition(
+                    builder.config.extended
+                        && builder
+                            .config
+                            .tools
+                            .as_ref()
+                            .map_or(true, |tools| tools.iter().any(|tool| $tool_name.contains(tool))),
```

Comment:
Hm, I think we should be able to go for exact match over contains? Is there a reason you opted for contains?

Reply:
It's because Clippy is called `clippy` in the config: https://github.com/rust-lang/rust/blob/4ca5fd2d7b6b1d75b6cb8f679e8523fb3e7b19e2/config.toml.example#L200
And `clippy-driver` in tools.rs: https://github.com/rust-lang/rust/blob/4ca5fd2d7b6b1d75b6cb8f679e8523fb3e7b19e2/src/bootstrap/tool.rs#L643

`tool_name` argument of `tool_extended` macro has to match binary name and Clippy is actually `clippy-driver`.
Changing `clippy` to `clippy-driver` in the config file doesn't seem right and it's unnecessary breaking change.

Reply:
I'd prefer to instead match specifically on clippy/clippy-driver and handle them separately.